includes/data-port/class-sensei-data-port-manager.php

```diff
@@ -0,0 +1,201 @@
+<?php
+/**
+ * File containing the Sensei_Data_Port_Manager class.
+ *
+ * @package sensei
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * This class is responsible creating, deleting and keeping track of data port jobs.
+ */
+class Sensei_Data_Port_Manager implements JsonSerializable {
+	const OPTION_NAME = 'sensei-tools-jobs';
+
+	/**
+	 * An array of all in progress data port jobs. It has the following format:
+	 * {
+	 *
+	 *     @type string $id        Unique id for this job.
+	 *     @type int    $user_id   The user which initiatied this job.
+	 *     @type int    $time      When the job started.
+	 *     @type string $handler   The class which handles this job.
+	 * }
+	 *
+	 * @var array
+	 */
+	private $data_port_jobs;
+
+	/**
+	 * Tracks if the data port jobs have been updated.
+	 *
+	 * @var boolean
+	 */
+	private $has_changed;
+
+	/**
+	 * Instance of singleton.
+	 *
+	 * @var self
+	 */
+	private static $instance;
+
+	/**
+	 * Fetches an instance of the class.
+	 *
+	 * @return self
+	 */
+	public static function instance() {
+
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Sensei_Data_Port_Manager constructor.
+	 */
+	private function __construct() {
+		$json_string       = get_option( self::OPTION_NAME );
+		$this->has_changed = false;
+
+		if ( $json_string ) {
+			$this->data_port_jobs = json_decode( $json_string, true );
+		} else {
+			$this->data_port_jobs = [];
+		}
+	}
+
+	/**
+	 * Initializes the data port manager.
+	 */
+	public function init() {
+		foreach ( $this->data_port_jobs as $job ) {
+			add_action( Sensei_Data_Port_Job::get_option_name( $job['id'] ), [ $this, 'run_data_port_job' ] );
```

Comment:
It seems weird to me that we're using the option name as an action name. Is this a common practice? Would it be better to create the action name independently and inject the Job ID?

Reply:
I have added a constant for the hook name in 84ef9fba9ee2c2a46a8a8924a92edc7f300693ac. 

The idea is that every currently running data port job will schedule an action with the same name. Then in the manager we add an action for all running jobs and by using the job id every triggered action will call run for the according job. Since each action will use different args the jobs should not mess with each other. This is something that we will need to test when we implement an actual job though.

Reply:
Hmm. Yes, I guess this is still a bit theoretical before we actually have code using this. I just didn't like this tight coupling between the option name and the job name. Seems like they should be independent from each other.